### PR TITLE
Adding match testing

### PR DESCRIPTION
**Pull request by cedi95**

### CODE REVIEW

File: src/test/matching.test.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+const postInfoShouldntMatch = {
+  userId: authUser573.uid,
+  title: 'Hiking',
+  location: {
+    latitude: parseFloat(67.23563352505248),
+    longitude: parseFloat(9.845367431640627),
+  },
+  category: '0',
+};
+const pinStatus = { ready1: false, ready2: false, ready3: false };
```
Comment (ochysp) [inline src/test/matching.test.js]: useless pinStatus

File: src/test/matching.test.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+const postStatus = {
```
Comment (ochysp) [inline src/test/matching.test.js]: useless postStatus

File: src/test/matching.test.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+});
+
+beforeEach(() => {
+  haltIfLiveDB();
+  deleteTestDbOnRootLevel();
+  setUpForUse();
+});
+
+describe('Test matching', () => {
+  describe('#checks Listing', () => {
```
Comment (ochysp) [inline src/test/matching.test.js]: Why is there a hashtag?

File: src/test/matching.test.js [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+    it('should get matches of two different categories', () => {
+      const PinsOfUser123 = mount(<MyPins authUser={authUser123} />);
+      PinsOfUser123.find('ListItem').first().simulate('click');
+      expect(PinsOfUser123.containsAllMatchingElements([
+        <h3>
+          Bike-Tour
```
Comment (ochysp) [inline src/test/matching.test.js]: Duplicate definition of text

File: src/test/matching.test.js [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    longitude: parseFloat(9.845367431640627),
+  },
+  category: '0',
+};
+const pinStatus = { ready1: false, ready2: false, ready3: false };
+const postStatus = {
+  ready1: false, ready2: false, ready3: false, ready4: false,
+};
+
+const setUpForUse = () => {
```
Comment (ochysp) [inline src/test/matching.test.js]: copy paste code, could be written as a loop

File: src/test/matching.test.js (hunk 3 above)
Comment (cedi95) [reply src/test/matching.test.js]: because you did it in the first place

File: src/test/matching.test.js (hunk 1 above)
Comment (cedi95) [reply src/test/matching.test.js]: you are useless

File: src/test/matching.test.js (hunk 2 above)
Comment (cedi95) [reply src/test/matching.test.js]: ok well written comment of yours

File: src/test/matching.test.js (hunk 4 above)
Comment (cedi95) [reply src/test/matching.test.js]: I hope you like my changes then

File: src/test/matching.test.js (hunk 5 above)
Comment (cedi95) [reply src/test/matching.test.js]: as your shitty comments

File: src/test/matching.test.js (hunk 3 above)
Comment (ochysp) [reply src/test/matching.test.js]: That was not intentional. Can be deleted.

**Pull request merged**